### ISSUE TITLE
mac_users implementation of chhome() doesn't know what to do with "persist" argument (broken in both 2015.8.7 and develop)

### ISSUE THREAD

**Issue by nshalman** [saltstack/salt#31409]
Traceback:
```
          ID: create_mac_user
    Function: user.present
        Name: foobar
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/local/salt/lib/python2.7/site-packages/salt/state.py", line 1624, in call
                  **cdata['kwargs'])
                File "/usr/local/salt/lib/python2.7/site-packages/salt/loader.py", line 1491, in wrapper
                  return f(*args, **kwargs)
                File "/usr/local/salt/lib/python2.7/site-packages/salt/states/user.py", line 471, in present
                  __salt__['user.chhome'](name, val, True)
              TypeError: chhome() takes exactly 2 arguments (3 given)
```

This is the broken code:
https://github.com/saltstack/salt/blob/v2015.8.7/salt/modules/mac_user.py#L269-L294

Presumably non-broken implementations here:
https://github.com/saltstack/salt/blob/v2015.8.7/salt/modules/pw_user.py#L228-L249
https://github.com/saltstack/salt/blob/v2015.8.7/salt/modules/solaris_user.py#L243-L265

**Comment by jfindlay** [comment]
Thanks for reporting, @nshalman.

**Comment by terminalmage** [comment]
This is not the right way to go about this. The ``persist`` param is meant to implement a feature native to the user management tools in Linux/BSD/etc. The solution for platforms that do not support this feature out of the box isn't to try to manually implement it, that's just prone to errors. No, we should accept the parameter to ensure the states don't blow up, and then if necessary, log that an unsupported option was used. We do this in a number of places already, so that we can reconcile the different options/features native to different package managers, or CLI options specific to certain versions of git, etc.

I noticed we had something similar set up for Windows. We should never have done this, either.

I've opened https://github.com/saltstack/salt/pull/35901 to address this and to properly document the lack of support for this feature on Windows/Mac.
